papertty/papertty.py

```diff
@@ -410,7 +394,14 @@ def showtext(self, text, fill, cursor=None, portrait=False, flipx=False, flipy=F
                               self.white)
             # create the Draw object and draw the text
             draw = ImageDraw.Draw(image)
-            draw.text((0, 0), text, font=self.font, fill=fill, spacing=self.spacing)
+
+            # Split the text up by line and display each line individually.
+            # This is a workaround for a font height bug in PIL
+            lines = text.split('\n')
+            for i in range(self.rows):
+                line = lines[i] if i < len(lines) else ''
+                y = i * self.font_height
+                draw.text((0, y), line, font=self.font, fill=fill, spacing=self.spacing)
```

Comment:
It looks like this will unconditionally invoke `draw.text()` for every row even if there was only a single character we need to put on the screen. While it may or may not have a negligible impact on performance (mostly thinking about RPi Zero here), on a chronic combination of big display + tiny font this might do hundreds of unnecessary calls per update, so I'd maybe add something like this (since there's no point drawing an empty string):
```suggestion
                if line:
                     draw.text((0, y), line, font=self.font, fill=fill, spacing=self.spacing)
```

Reply:
Makes sense. I've pulled in your suggested change. Individual draws should be negligible, especially if they're empty, but there's no harm in skipping them since the image itself fills the screen anyway.
In that case there's no point iterating all the rows, just the lines with text on them. I've added a commit to that effect.